metadata-ingestion/src/datahub/ingestion/graph/client.py

```diff
@@ -465,41 +487,78 @@ def get_container_urns_by_filter(
             yield x["entity"]
 
     def get_urns_by_filter(
-        self, platform: str, batch_size: int = 10000
+        self,
+        *,
+        entity_types: Optional[List[str]] = None,
+        platform: Optional[str] = None,
+        batch_size: int = 10000,
     ) -> Iterable[str]:
+        """Fetch all urns that match the given filters.
+
+        Filters are combined conjunctively. If multiple filters are specified, the results will match all of them.
+        Note that specifying a platform filter will automatically exclude all entity types that do not have a platform.
+
+        :param entity_types: List of entity types to include. If None, all entity types will be returned.
+        :param platform: Platform to filter on. If None, all platforms will be returned.
+        """
+
+        types = None
+        if entity_types is not None:
+            if not entity_types:
+                raise ValueError("entity_types cannot be an empty list")
+
+            types = [_graphql_entity_type(entity_type) for entity_type in entity_types]
+
         # Does not filter on env, because env is missing in dashboard / chart urns and custom properties
         # For containers, use { field: "customProperties", values: ["instance=env}"], condition:EQUAL }
         # For others, use { field: "origin", values: ["env"], condition:EQUAL }
 
-        query = """
-        query scrollEntitiesForPlatform($platform: String!, $batchSize: Int!, $scrollId: String) {
-          scrollAcrossEntities(input: { query: "*", count:$batchSize,
-            scrollId: $scrollId,
-            orFilters: [
-              {and: [{
-                field: "platform.keyword",
-                values: [$platform],
-                condition: EQUAL,
-              }]}
+        andFilters = []
+        if platform:
+            andFilters += [
+                {
+                    "field": "platform.keyword",
+                    "values": [make_data_platform_urn(platform)],
+                    "condition": "EQUAL",
+                }
             ]
-          }) {
-            nextScrollId
-            searchResults {
-              entity {
-                urn
-              }
+        orFilters = [{"and": andFilters}]
+
+        query = textwrap.dedent(
+            """
+            query scrollUrnsWithFilters(
+                $types: [EntityType!],
+                $orFilters: [AndFilterInput!],
+                $batchSize: Int!,
+                $scrollId: String) {
+
+                scrollAcrossEntities(input: {
+                    query: "*",
+                    count: $batchSize,
+                    scrollId: $scrollId,
+                    types: $types,
+                    orFilters: $orFilters,
+                    searchFlags: { skipHighlighting: true }
```

Comment:
What's this do?

Reply:
skipHighlighting tells elasticsearch not to generate those "matched column xyz" snippets. It should yield a minor performance improvement